tracing/log.go

```diff
@@ -60,7 +64,50 @@ func (h *LogrusHook) Fire(entry *logrus.Entry) error {
 func logrusDataToAttrs(data logrus.Fields) []attribute.KeyValue {
 	attrs := make([]attribute.KeyValue, 0, len(data))
 	for k, v := range data {
-		attrs = append(attrs, attribute.Any(k, v))
+		attrs = append(attrs, any(k, v))
 	}
 	return attrs
 }
+
+func any(k string, v interface{}) attribute.KeyValue {
+	if v == nil {
+		return attribute.String(k, "<nil>")
+	}
+
+	rv := reflect.ValueOf(v)
+
+	switch rv.Kind() {
+	case reflect.Slice:
+		switch typed := v.(type) {
+		case []string:
+			return attribute.StringSlice(k, typed)
+		case []bool:
+			return attribute.BoolSlice(k, typed)
+		case []float64:
+			return attribute.Float64Slice(k, typed)
+		case []int64:
+			return attribute.Int64Slice(k, typed)
+		case []int:
+			return attribute.IntSlice(k, typed)
+		}
+	case reflect.Bool:
+		return attribute.Bool(k, rv.Bool())
+	case reflect.Int, reflect.Int8, reflect.Int16:
+		return attribute.Int(k, int(rv.Int()))
+	case reflect.Int64:
+		return attribute.Int64(k, rv.Int())
+	case reflect.Float64:
+		return attribute.Float64(k, rv.Float())
+	case reflect.String:
+		return attribute.String(k, rv.String())
+	}
+
+	if stringer, ok := v.(fmt.Stringer); ok {
+		return attribute.String(k, stringer.String())
+	}
+	if b, err := json.Marshal(v); b != nil && err == nil {
+		return attribute.String(k, string(b))
+	}
+
+	return attribute.String(k, v.(string))
```

Comment:
Doesn't this panic if `v` isn't string (e.g. func)?

Reply:
Thanks, I meant to take this out.

Reply:
Fixed this.